app/services/liquid_assigns_service.rb

```diff
@@ -15,4 +18,44 @@ def assigns
 
   def assigns_pro
   end
+
+  # This method uses Liquid::VariableLookup to find all liquid variables from
+  # a given text. We use the list to know which project assign we need.
+  def assigns_from_content
+    return AVAILABLE_PROJECT_ASSIGNS if text.nil?
+
+    variable_lookup = Liquid::VariableLookup.parse(text)
+    return (variable_lookup.lookups & AVAILABLE_PROJECT_ASSIGNS)
+  end
+
+  def cached_drops(records, record_type)
+    return [] if records.empty?
+
+    cache_key = "liquid-project-#{project.id}-#{record_type.pluralize}:#{records.maximum(:updated_at).to_i}-#{records.count}"
```

Comment:
it seems we're working hard here to craft a unique key. Have you looked into how the Rails code does this when you pass a collection in your view cache call? Meybe there's a method we can use that takes a prefix and a collection and spits the unique cache key?

Reply:
This is what I meant:
https://api.rubyonrails.org/classes/ActiveSupport/Cache.html#method-c-expand_cache_key